lightning_transformers/core/callback.py

```diff
@@ -11,11 +11,123 @@
+        if isinstance(trainer.logger, WANDBLogger):
+            trainer.logger.__init__(init_kwargs={"project": "lightning-transformers"})
+
+    def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        optimizer = trainer.optimizers
+
+        if len(optimizer) > 1:
+            raise MisconfigurationException("SparseML only supports training with one optimizer.")
+        optimizer = optimizer[0]
+
+        loggers = trainer.logger
+
+        if not isinstance(loggers, list):
+            loggers = [loggers]
+
+        self.manager.initialize(pl_module, epoch=0.0, logger=loggers)
+        self.manager.initialize_loggers(loggers)
+
+        optimizer = self.manager.modify(
+            pl_module, optimizer, steps_per_epoch=self._num_training_steps_per_epoch(trainer), epoch=0
+        )
+
+        trainer.optimizers = [optimizer]
+
+    @staticmethod
+    def export_to_sparse_onnx(
+        model: "LightningModule", output_dir: str, sample_batch: Optional[Tensor] = None, **kwargs
+    ) -> None:
+        """Exports the model to ONNX format."""
+        with model._prevent_trainer_and_dataloaders_deepcopy():
+            exporter = ModuleExporter(model.model, output_dir=output_dir)
+            sample_batch = sample_batch if sample_batch is not None else model.example_input_array
+            if sample_batch is None:
+                raise MisconfigurationException(
+                    "To export the model, a sample batch must be passed via "
+                    "``SparseMLCallback.export_to_sparse_onnx(model, output_dir, sample_batch=sample_batch)`` "
+                    "or an ``example_input_array`` property within the LightningModule"
+                )
+
+            # the following is adapted from @natuan and @spacemanidol
+            sess = None
+            num_samples = 0
+
+            sample_inputs = os.path.join(output_dir, "sample-inputs")
+            sample_outputs = os.path.join(output_dir, "sample-outputs")
+            os.makedirs(sample_inputs, exist_ok=True)
+            os.makedirs(sample_outputs, exist_ok=True)
+
+            if sess is None:
+                forward_args_spec = inspect.getfullargspec(exporter._module.__class__.forward)
+                one_sample_input = collections.OrderedDict([(f, sample_batch[f][0].long().reshape(1, -1))
+                                                            for f in forward_args_spec.args if f in sample_batch])
+
+                try:
+                    exporter.export_onnx(sample_batch=one_sample_input, convert_qat=True, **kwargs)
+                    exporter.export_onnx(
+                        sample_batch=one_sample_input,
+                        name="small_model.onnx",
+                        convert_qat=True,
+                        export_params=False,
+                        **kwargs,
+                    )
+                    onnx_file = os.path.join(output_dir, "model.onnx")
+
+                except Exception:
```

Comment:
pls be more specific what exception is expected here

Reply:
changed to RunTimeError